compiler/src/dotty/tools/dotc/transform/localopt/FormatChecker.scala

```diff
@@ -209,17 +215,20 @@ class TypedFormatChecker(partsElems: List[Tree], parts: List[String], args: List
     def accepts(arg: Type): Boolean =
       kind match
         case BooleanXn  => arg == defn.BooleanType orElse warningAt(CC)("Boolean format is null test for non-Boolean")
-        case IntegralXn =>
-          arg == BigIntType || !cond(cc) {
-            case 'o' | 'x' | 'X' if hasAnyFlag("+ (") => "+ (".filter(hasFlag).foreach(bad => badFlag(bad, s"only use '$bad' for BigInt conversions to o, x, X")) ; true
-          }
+        case IntegralXn => arg == BigIntType || !cond(cc) {
+          case 'o' | 'x' | 'X' if hasAnyFlag("+ (") => "+ (".filter(hasFlag).foreach(bad => badFlag(bad, s"only use '$bad' for BigInt conversions to o, x, X")); true
+        }
         case _ => true
 
+    def lintToString(arg: Type): Unit =
+      if ctx.settings.Whas.toStringInterpolated && kind == StringXn && !(arg.widen =:= defn.StringType) && !arg.isPrimitiveValueType
+      then warningAt(CC)("interpolation uses toString")
```

Comment:
This would probably be better, when done as a separate PR, but it might be useful to add message ids for all the error/warning messages in this file.